src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/settings/SettingsPluginConfigurable.kt

```diff
@@ -68,9 +68,23 @@ class SettingsPluginConfigurable : Configurable {
         settingsState.buildPath = settingsComponent!!.buildPath
         settingsState.buildCommand = settingsComponent!!.buildCommand
         settingsState.telemetryEnabled = settingsComponent!!.telemetryEnabled
+        if (checkEmptyTelemetryPath(settingsState)) return
         settingsState.telemetryPath = settingsComponent!!.telemetryPath
     }
 
+    /**
+     * Check if the telemetry path is empty when telemetry is enabled.
+     * If empty, then sets to previous state. Else, keep the new one.
+     */
+    private fun checkEmptyTelemetryPath(settingsState: TestGenieSettingsState): Boolean {
+        if (settingsComponent!!.telemetryEnabled && settingsComponent!!.telemetryPath.isBlank()) {
+            settingsState.telemetryPath = settingsState.telemetryPath
+            settingsComponent!!.telemetryPath = settingsState.telemetryPath
+            return true
+        }
+        return false
+    }
+
```

Comment:
Why do you need this check?

Reply:
As KDoc on top of the function says, this method is responsible for checking if the telemetry path field is empty, when the telemetry checkbox is enabled. If the path is empty and the user triggered the "apply" or "ok" buttons then the path will be reset to the previous saved path, if the path was entered, then it will be updated to the current one.

Reply:
I see.